framework/src/utils/PetscSupport.C

```diff
@@ -511,7 +557,16 @@ petscSetDefaults(FEProblemBase & problem)
     PetscLinearSolver<Number> * petsc_solver = dynamic_cast<PetscLinearSolver<Number> *>(
         lin_sys.linearImplicitSystem().get_linear_solver());
     KSP ksp = petsc_solver->ksp();
-    petscSetKSPDefaults(problem, ksp);
+
+    if (problem.hasLinearConvergenceObjects())
+    {
+      LibmeshPetscCallA(
+          lin_sys.comm().get(),
+          KSPSetConvergenceTest(ksp, petscLinearConverged, &problem, LIBMESH_PETSC_NULLPTR));
+    }
+
+    // We dont set the KSP defaults here because they seem to clash with the linear solve parameters
+    // set in FEProblemBase::solveLinearSystem
```

Comment:
I guess this must work, but I don't understand it. So there is no default linear convergence object being added (`problem.hasLinearConvergenceObjects() == false)`, so in most cases today, it just looks like `petscSetKSPDefaults(problem, ksp)` has no effect (we could delete this line without any of your other changes)?

By the way, is this only for the `solve_type = LINEAR` solves, or also for the linear solves inside nonlinear solves?